devices/HapticGlove/src/HapticGlove.cpp

```diff
@@ -305,6 +305,14 @@ bool HapticGlove::open(yarp::os::Searchable& config)
 
     m_pImpl->hapticActuatorPrefix = getWearableName() + actuator::IHaptic::getPrefix();
 
+    for (size_t i = 0; i < m_pImpl->gloveData.humanFingerNames.size(); i++) {
+        m_pImpl->sensegloveHapticActuatorVector.push_back(
+            std::make_shared<SenseGloveImpl::SenseGloveHapticActuator>(
+                m_pImpl.get(),
```

Comment:
This is fishy. Why do you need the raw pointer to a private implementation?

Reply:
well that one is written in this way to be consistent with other parts of the code. For more details see the SenseGloveHapticActuator class.

Reply:
Still, it is fishy. Consider the private implementation as a key to your locker. You should be the only owner of your locker, but you are giving around copies of your key. It is not really safe.

Reply:
An issue to discuss and fix this is opened: https://github.com/ami-iit/element_haptic-gloves/issues/192